api/v1beta1/cloudstackisolatednetwork_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// The presence of a finalizer prevents CAPI from deleting the corresponding CAPI data.
+	IsolatedNetworkFinalizer = "cloudstackisolatednetwork.infrastructure.cluster.x-k8s.io"
+)
+
+// CloudStackIsolatedNetworkSpec defines the desired state of CloudStackIsolatedNetwork
+type CloudStackIsolatedNetworkSpec struct {
+	// Cloudstack Network ID the cluster is built in.
+	// +optional
+	ID string `json:"id,omitempty"`
+
+	// Cloudstack Network Type the cluster is built in.
+	// + optional
+	Type string `json:"type,omitempty"`
+
+	// Cloudstack Network Name the cluster is built in.
+	// +optional
+	Name string `json:"name"`
+
+	// The kubernetes control plane endpoint.
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// IdentityRef is a reference to a identity to be used when reconciling this cluster
+	// +optional
+	// +k8s:conversion-gen=false
+	IdentityRef *CloudStackIdentityReference `json:"identityRef,omitempty"`
```

Comment:
Is this used?  I couldn't find a reference to it elsewhere.

Reply:
Nope, will remove.